### PR TITLE
[Concept Entry] Java / Boolean Logic

### LINKED ISSUE #4175: [Entry] Java: Java/ Boolean Logic
### Reason/inspiration (optional)

Add the topic of boolean logic.
For example: what !false, &&, ||, true, false, if(boolean variable) in as a statement means.

- Use a codebyte to show the example 
- define what each symbol does

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [x] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by noahpgordon**
### Description

Added **boolean-logic.md** within Java. Closes #4175 

(I'm unsure if classes within Java Codebytes must have specific names to work. Please advise.)

### Type of Change

- Adding a new entry

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/java/concepts/boolean-logic/boolean-logic.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  - 'Conditionals'
+  - 'Java'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/introduction-to-android-with-java'
+---
+
+One of the primitive [data types](https://www.codecademy.com/resources/docs/java/data-types) in Java is the `boolean`. A boolean object takes a value of `true` or `false`.
+
+**Boolean logic** describes the ways in which boolean values can be combined and manipulated. Java implements boolean logic through a set of [operators](https://www.codecademy.com/resources/docs/java/operators) and [methods](https://www.codecademy.com/resources/docs/java/methods). These are described below.
```
Comment (dakshdeepHERE) [inline content/java/concepts/boolean-logic/boolean-logic.md:20]:
```suggestion
**Boolean logic** describes how boolean values can be combined and manipulated. Java implements boolean logic through a set of [operators](https://www.codecademy.com/resources/docs/java/operators) and [methods](https://www.codecademy.com/resources/docs/java/methods).
```

File: content/java/concepts/boolean-logic/boolean-logic.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+|-|-|-|
+| `logicalAnd(boolean A, boolean B)` | Returns the result of `(A && B)` | AND / Conjunction |
+| `logicalOr(boolean A, boolean B)` | Returns the result of `(A \|\| B)` | OR / Inclusive disjunction |
+| `logicalXor(boolean A, boolean B)` | Returns the result of `(A ^ B)` | XOR / Exculsive disjunction |
+
+## Use in Conditionals
+
+Boolean logic is commonly used within conditionals to control the flow of a program.
+
+### Codebyte Example
```
Comment (dakshdeepHERE) [inline content/java/concepts/boolean-logic/boolean-logic.md:47]: Lets make this is a simple example section. As our online compiler doesn't support java.

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
I have requested few changes @noahpgordon 🚀

**Comment by noahpgordon** [comment]
@dakshdeepHERE Implemented.

**Review by dakshdeepHERE — APPROVED** [review]
LGTM 🚀

File: content/java/concepts/boolean-logic/boolean-logic.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
+---
+Title: 'Boolean Logic'
+Description: 'A set of operators and methods in Java for manipulating booleans.'
+Subjects:
+  - 'Computer Science'
```
Comment (avdhoottt) [inline content/java/concepts/boolean-logic/boolean-logic.md:5]:
```suggestion
  - 'Computer Science'
  - 'Code Foundations'
```

**Review by avdhoottt — COMMENTED** [review]
Looks to get merged!!

**Review by avdhoottt — APPROVED** [review]

**Pull request merged**